Source/ASButtonNode.mm

```diff
@@ -301,12 +350,17 @@ - (void)setImageAlignment:(ASButtonNodeImageAlignment)imageAlignment
 #if TARGET_OS_IOS
 - (void)setTitle:(NSString *)title withFont:(UIFont *)font withColor:(UIColor *)color forState:(UIControlState)state
 {
-  NSDictionary *attributes = @{
-    NSFontAttributeName: font ? : [UIFont systemFontOfSize:[UIFont buttonFontSize]],
-    NSForegroundColorAttributeName : color ? : [UIColor blackColor]
-  };
-    
-  NSAttributedString *string = [[NSAttributedString alloc] initWithString:title attributes:attributes];
+  NSMutableDictionary *attributes = [NSMutableDictionary dictionary];
+  attributes[NSFontAttributeName] = font ? : [UIFont systemFontOfSize:[UIFont buttonFontSize]];
+  if (color != nil) {
+    // From apple's documentation: If color is not specified, NSForegroundColorAttributeName will fallback to black
+    // Only set if the color is nonnull
+    attributes[NSForegroundColorAttributeName] = color;
+  } else if (self.tintColor != nil) {
+    attributes[NSForegroundColorAttributeName] = self.tintColor;
+  }
```

Comment:
Do we need a fallback like before in the case that `color` and `tintColor` are nil?

Reply:
I don't believe so, if you look at the snapshot tests you'll see that when we specify `nil` as the background color it ends up being black anyways which preserves the existing logic